### ISSUE TITLE
Virtual network rules property is not updated during ARM deployment.

### ISSUE THREAD

**Issue by xenalite** [Azure/acr#246]
**Describe the bug**
Virtual network rules property is not updated during ARM deployment.

**To Reproduce**
1. Have a Premium SKU registry
2. Perform ARM deployment to update `properties.networkRuleSet.virtualNetworkRules` property to link with some subnets.
3. The property is not updated after deployment, whereas it is after Azure CLI or Azure Portal update.

**Expected behavior**
ARM deployment to yield the same results as Azure CLI or Azure Portal.

**Templates**

[template.txt](https://github.com/Azure/acr/files/3338285/template.txt)
[parameters.txt](https://github.com/Azure/acr/files/3338287/parameters.txt)

Properties applied:
```json
{
  "loginServer":"ixsamlcem20cr01dev.azurecr.io",
  "creationDate":"2019-06-24T09:57:02.6817129Z",
  "provisioningState":"Succeeded",
  "adminUserEnabled":false,
  "networkRuleSet":{
    "defaultAction":"Deny",
    "ipRules":[
      {
        "action":"Allow",
        "value":"104.209.106.12"
      },
      {
        "action":"Allow",
        "value":"104.209.107.12"
      },
      {
        "action":"Allow",
        "value":"104.210.236.12"
      },
      {
        "action":"Allow",
        "value":"104.210.237.12"
      },
      {
        "action":"Allow",
        "value":"32.65.74.146"
      },
      {
        "action":"Allow",
        "value":"32.65.74.147"
      },
      {
        "action":"Allow",
        "value":"32.65.74.148"
      },
      {
        "action":"Allow",
        "value":"32.65.74.149"
      },
      {
        "action":"Allow",
        "value":"32.65.74.150"
      },
      {
        "action":"Allow",
        "value":"158.82.143.130"
      },
      {
        "action":"Allow",
        "value":"158.82.159.130"
      },
      {
        "action":"Allow",
        "value":"202.167.233.229"
      },
      {
        "action":"Allow",
        "value":"27.111.213.117"
      }
    ],
    "virtualNetworkRules":[
      {
        "id":"/subscriptions/de0f5db1-e28d-4f5f-9f32-7ba414dd330d/resourceGroups/aml-dev-cluster-westeurope/providers/Microsoft.Network/virtualNetworks/IXS-AML-C-EM20-VNET-01-DEV/subnets/Subnet-01",
        "action":"Allow"
      },
      {
        "id":"/subscriptions/de0f5db1-e28d-4f5f-9f32-7ba414dd330d/resourceGroups/aml-dev-cluster-westeurope/providers/Microsoft.Network/virtualNetworks/IXS-AML-C-EM20-VNET-01-DEV/subnets/Subnet-02",
        "action":"Allow"
      }
    ]
  }
}
```

**Any relevant environment information**
- Registry name: ixsamlcem20cr01dev
- Correlation ID: 68ff8a4d-dad6-4525-9191-7da66bc89d6a

It is similar to issue #232 but this time I'm pretty sure the JSON schema is correct.

**Comment by xenalite** [comment]
Actually scratch that. Azure CLI doesn't work either! It worked once yesterday. Today I tried updating rules with subnets 5 times and the changes did not get applied.

**Comment by xenalite** [comment]
Seems this person gets the same issue as me: 
https://feedback.azure.com/forums/903958-azure-container-registry/suggestions/37475404-adding-virtual-networks-via-portal-or-cli-fails

**Comment by yuwaMSFT2** [comment]
Have you had the service endpoint enabled for container registry for your VNET first?

Service endpoint for subnets
https://docs.microsoft.com/en-us/azure/container-registry/container-registry-vnet#service-endpoint-for-subnets
To allow access from a subnet in a virtual network, you need to add a service endpoint for the Azure Container Registry service.

**Comment by xenalite** [comment]
@yuwaMSFT2 Thanks I was pretty sure I enabled the service endpoints on the subnets ahead of the time.
When I executed Azure CLI calls that did not work, I tried to join ACR to all subnets, including ones without Microsoft.ContainerRegistry endpoint enabled. When I dropped down my request to only include subnets with service endpoint enabled, the change was applied correctly.
This worked with ARM templates and with Azure SDK for .NET.

I think my template from OP did not work because despite both subnets having service endpoints enabled, I had an "update" template running against subnets at the same time as updating ACR. When ACR was attempting to join to subnets, the subnets were in "Updating" state. Possibly a race condition there.

The issue is resolved, but I would leave preview feedback that there should be warnings that service endpoints are not enabled on the subnet, like for other resources that can be joined to subnets.
